app/Controllers/PrestitiController.php

```
@@ -858,15 +858,35 @@ public function update(Request $request, Response $response, mysqli $db, int $id
+            $oldScadenza = (string) $current['data_scadenza'];
+            if ($newPrestito !== $oldPrestito || $newScadenza !== $oldScadenza) {
+                $claimedWindows = [];
+                if ($newPrestito < $oldPrestito) {
+                    // Y-m-d strings compare correctly lexicographically. The inclusive
+                    // boundary day (old start / old due) mirrors renew()'s convention.
+                    $claimedWindows[] = [$newPrestito, min($oldPrestito, $newScadenza)];
+                }
+                if ($newScadenza > $oldScadenza) {
+                    $claimedWindows[] = [max($oldScadenza, $newPrestito), $newScadenza];
+                }
                 $capacity = new \App\Services\CapacityService($db);
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '884', column '33'). (undefined)

(MissingImport)